experimental/traffic-portal/src/app/core/topologies/topology-details/topology-details.component.ts

```diff
@@ -0,0 +1,152 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { NestedTreeControl } from "@angular/cdk/tree";
+import { Location } from "@angular/common";
+import { Component, OnInit } from "@angular/core";
+import { MatDialog } from "@angular/material/dialog";
+import { MatTreeNestedDataSource } from "@angular/material/tree";
+import { ActivatedRoute } from "@angular/router";
+import { ResponseTopology } from "trafficops-types";
+
+import { TopologyService, TopTreeNode } from "src/app/api";
+import {
+	DecisionDialogComponent,
+	DecisionDialogData,
+} from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import {
+	NavigationService
+} from "src/app/shared/navigation/navigation.service";
+
+/**
+ * TopologyDetailComponent is the controller for a Topology's "detail" page.
+ */
+@Component({
+	selector: "tp-topology-details",
+	styleUrls: ["./topology-details.component.scss"],
+	templateUrl: "./topology-details.component.html",
+})
+export class TopologyDetailsComponent implements OnInit {
+	public new = false;
+	public topology: ResponseTopology = {
+		description: "",
+		lastUpdated: new Date(),
+		name: "",
+		nodes: [],
+	};
+	public showErrors = false;
+	public topologies: Array<ResponseTopology> = [];
+	public topologySource = new MatTreeNestedDataSource<TopTreeNode>();
+	public topologyControl = new NestedTreeControl<TopTreeNode>(node => node.children);
+
+	constructor(
+		private readonly route: ActivatedRoute,
+		private readonly api: TopologyService,
+		private readonly location: Location,
+		private readonly dialog: MatDialog,
+		private readonly navSvc: NavigationService,
+	) {
+	}
+
+	/**
+	 * Angular lifecycle hook where data is initialized.
+	 */
+	public async ngOnInit(): Promise<void> {
+		const name = this.route.snapshot.paramMap.get("name");
+		if (name === null) {
+			console.error("missing required route parameter 'name'");
+			return;
+		}
+
+		const topologiesPromise = this.api.getTopologies().then(topologies => this.topologies = topologies);
+		if (name === "new") {
```

Comment:
This means I can't ever create/manipulate a Topology named "new" in this UI - you can look at the server capabilities page source to see how we've worked around this before.